jib-core/src/test/java/com/google/cloud/tools/jib/cache/LayerEntriesSelectorTest.java

```diff
@@ -82,10 +93,25 @@ public void testGenerateSelector_empty() throws IOException {
   @Test
   public void testGenerateSelector() throws IOException {
     DescriptorDigest expectedSelector =
-        JsonTemplateMapper.toBlob(toLayerEntryTemplates(IN_ORDER_LAYER_ENTRIES))
+        JsonTemplateMapper.toBlob(toLayerEntryTemplates(inOrderLayerEntries))
             .writeTo(ByteStreams.nullOutputStream())
             .getDigest();
     Assert.assertEquals(
-        expectedSelector, LayerEntriesSelector.generateSelector(OUT_OF_ORDER_LAYER_ENTRIES));
+        expectedSelector, LayerEntriesSelector.generateSelector(outOfOrderLayerEntries));
+  }
+
+  @Test
+  public void testGenerateSelector_fileModified() throws IOException, InterruptedException {
+    Path layerFile = temporaryFolder.newFolder("testFolder").toPath().resolve("file");
+    Files.write(layerFile, "hello".getBytes(StandardCharsets.UTF_8));
+    LayerEntry layerEntry = new LayerEntry(layerFile, AbsoluteUnixPath.get("/extraction/path"));
+    DescriptorDigest expectedSelector =
+        LayerEntriesSelector.generateSelector(ImmutableList.of(layerEntry));
+
+    // Verify that changing modified time generates a different selector
+    Thread.sleep(1000);
+    Files.write(layerFile, "goodbye".getBytes(StandardCharsets.UTF_8));
+    Assert.assertNotEquals(
+        expectedSelector, LayerEntriesSelector.generateSelector(ImmutableList.of(layerEntry)));
```

Comment:
Could also just check that setting last modified back to the original generates the same selector.